test/io/test_tim_grabber.cpp

```diff
@@ -0,0 +1,121 @@
+#include <pcl/test/gtest.h>
+#include <pcl/point_types.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/io/tim_grabber.h>
+
+#include <string>
+
+using PointT = pcl::PointXYZ;
+using CloudT = pcl::PointCloud<PointT>;
+
+std::string packets_dir_;
+std::string pcds_dir_;
+std::vector<std::string> packets_;
+std::vector<CloudT::ConstPtr> pcds_;
+std::vector<std::string> pcd_files_;
+
+class TestableTimGrabber : public pcl::TimGrabber {
+  using TimGrabber::TimGrabber;
+  public: using TimGrabber::point_cloud_xyz_ptr_;
+  public: using TimGrabber::processTimPacket;
+  public: using TimGrabber::updateLookupTables;
+  public: using TimGrabber::toPointClouds;
+};
+
+TEST (PCL, TestableTimGrabber)
```

Comment:
Please add an expected test:
* Create a `SetUp` where you create 2 things:
  * a point cloud (say a line at 45 degree from -135 to 135 degree)
  * a string with those distance values in hex (basically the format)
* Check that the 2 are equivalent

This will test that the code adheres to some "API". Your code requires a ground truth pcd to be present (and if you created the pcd using your code, we'd be testing your implementation, not the actual protocol)